src/components/chart/statistics-chart.ts

```diff
@@ -0,0 +1,315 @@
+        )}
+      </div>`;
+    }
+
+    if (!this.statisticsData || !Object.keys(this.statisticsData).length) {
+      return html`<div class="info">
+        ${this.hass.localize(
+          "ui.components.statistics_charts.no_statistics_found"
+        )}
+      </div>`;
+    }
+
+    return html`
+      <ha-chart-base
+        .data=${this._chartData}
+        .options=${this._chartOptions}
+        .plugins=${this._chartPlugins}
+        .chartType=${this.chartType}
+      ></ha-chart-base>
+    `;
+  }
+
+  private _createOptions() {
+    this._chartOptions = {
+      parsing: false,
+      animation: false,
+      scales: {
+        x: {
+          type: "time",
+          adapters: {
+            date: {
+              locale: this.hass.locale,
+            },
+          },
+          ticks: {
+            maxRotation: 0,
+            sampleSize: 5,
+            autoSkipPadding: 20,
+            major: {
+              enabled: true,
+            },
+            font: (context) =>
+              context.tick && context.tick.major
+                ? ({ weight: "bold" } as any)
+                : {},
+          },
+          time: {
+            tooltipFormat: "datetimeseconds",
+          },
+        },
+        y: {
+          ticks: {
+            maxTicksLimit: 7,
+          },
+        },
+      },
+      plugins: {
+        tooltip: {
+          mode: "nearest",
+          callbacks: {
+            label: (context) => `${context.dataset.label}: ${context.parsed.y}`,
+          },
+        },
+        filler: {
+          propagate: true,
+        },
+        legend: {
+          display: false,
+          labels: {
+            usePointStyle: true,
+          },
+        },
+      },
+      hover: {
+        mode: "nearest",
+      },
+      elements: {
+        line: {
+          tension: 0.4,
+          borderWidth: 1.5,
+        },
+        point: {
+          hitRadius: 5,
+        },
+      },
+    };
+  }
+
+  private _generateData() {
+    let colorIndex = 0;
+    const statisticsData = Object.values(this.statisticsData);
+    const datasets: ChartDataset<"line">[] = [];
+    let endTime: Date;
+
+    if (statisticsData.length === 0) {
+      return;
+    }
+
+    endTime =
+      this.endTime ||
+      // Get the highest date from the last date of each device
+      new Date(
+        Math.max(
+          ...statisticsData.map((stats) =>
+            new Date(stats[stats.length - 1].start).getTime()
+          )
+        )
+      );
+
+    if (endTime > new Date()) {
+      endTime = new Date();
+    }
+
+    const names = this.names || {};
+    statisticsData.forEach((stats) => {
+      const firstStat = stats[0];
+      let name = names[firstStat.statistic_id];
+      if (!name) {
+        const entityState = this.hass.states[firstStat.statistic_id];
+        if (entityState) {
+          name = computeStateName(entityState);
+        } else {
+          name = firstStat.statistic_id;
+        }
```

Comment:
```suggestion
        name = entityState ? computeStateName(entityState) : firstStat.statistic_id;
```
Alternative :man_shrugging: